Feature/dynamic module load

bigbitbus opened the pull request:
# Description
Load Opta modules dynamically instead of loading all of them in the layer.py file.

# Safety checklist
* [X] This change is backwards compatible and safe to apply by existing users
* [X] This change will NOT lead to data loss
* [X] This change will NOT lead to downtime who already has an env/service setup

## How has this change been tested, beside unit tests?
Manual testing

rabbitfang commented on opta/registry.py:
```diff
@@ -50,7 +51,7 @@ def make_registry_dict() -> Dict[Any, Any]:
         cloud_dict["modules"] = {**_make_module_registry_dict(module_path, alt_cloudname)}
         registry_dict[cloud] = cloud_dict
-
+    open("/tmp/registry.json", "w").write(json.dumps(registry_dict))  # nosec
```
Was this meant to be left in?

bigbitbus replied:
no, it was debug info, removed.

rabbitfang commented on opta/layer.py:
```diff
@@ -75,6 +35,55 @@
 from opta.utils import deep_merge, hydrate, logger, yaml
 from opta.utils.dependencies import validate_installed_path_executables
 
+PROCESSOR_DICT: Dict[str, str] = {
```
Can you please add a unit test that checks that all of the processors in this dict exist? Something like:
```python
for processor in PROCESSOR_DICT.values():
    pymodule_path = generate_pymodule_path(processor)
    importlib.import_module(pymodule_path)
```

bigbitbus replied:
Added a test to check if each module in the dictionary is actually in the directories.

bigbitbus commented:
@rabbitfang addressed the comments in the review. @juandiegopalomino as discussed when we use this in the new module api we will try to automatically infer the mapping instead of the dictionary. For now with @rabbitfang suggestion I added a test to fail if the module code does not exist at the expected location in the module directory.

rabbitfang commented on tests/test_layer.py:
```diff
@@ -322,3 +376,11 @@ def test_child(self, mocker: MockFixture):
+        for opta_module, processor in PROCESSOR_DICT.items():
+            module_class = get_processor_class(opta_module)
+            assert module_class.__name__ == processor
+        print("done")
```
```suggestion
```

rabbitfang reviewed (approved):
Everything else LGTM

bigbitbus commented:
@rabbitfang added your commit.
